fdbserver/TLogServer.actor.cpp

```diff
@@ -1526,14 +1530,19 @@ std::deque<std::pair<Version, LengthPrefixedStringRef>>& getVersionMessages(Refe
 	return tagData->versionMessages;
 };
 
-ACTOR Future<Void> waitForMessagesForTag(Reference<LogData> self, TLogPeekRequest* req) {
+ACTOR Future<Void> waitForMessagesForTag(Reference<LogData> self, TLogPeekRequest* req, double timeout) {
+	self->blockingPeeks += 1;
 	auto tagData = self->getTagData(req->tag);
-	if (tagData.isValid()) {
+	if (tagData.isValid() && !tagData->versionMessages.empty() && tagData->versionMessages.back().first > req->begin) {
```

Comment:
I believe versionMessages deque will be non empty only when actual data has been enqueued onto it. Empty messages will not be added to the deque. So if `waitForMessagesForTag` is called and empty messages have been received, but data has not been received, we will wait here for at most a full second. Is this your understanding too?

Reply:
Yes.

And,  the if-condition needs to be "tagData->versionMessages.back().first >= req->begin" (per Jingyu's comment in https://github.com/apple/foundationdb/pull/5058/commits/95e85aaffba7e7ad8dad85774ac7370f23a1e769). Will address that.

Reply:
Done.